neo/framework/Game.h

```diff
@@ -239,8 +239,87 @@ class idProgram;
+	virtual const idMD5Anim *	ANIM_GetAnim( const char *fileName ) = 0;
+	virtual int					ANIM_GetLength( const idMD5Anim *anim ) = 0;
+	virtual int					ANIM_GetNumFrames( const idMD5Anim *anim ) = 0;
+	virtual void				ANIM_CreateAnimFrame( const idRenderModel *model, const idMD5Anim *anim, int numJoints, idJointMat *frame, int time, const idVec3 &offset, bool remove_origin_offset ) = 0;
+	virtual idRenderModel *		ANIM_CreateMeshForAnim( idRenderModel *model, const char *classname, const char *animname, int frame, bool remove_origin_offset ) = 0;
+
+	// Articulated Figure calls for AF editor and Radiant.
+	virtual bool				AF_SpawnEntity( const char *fileName ) = 0;
+	virtual void				AF_UpdateEntities( const char *fileName ) = 0;
+	virtual void				AF_UndoChanges( void ) = 0;
+	virtual idRenderModel *		AF_CreateMesh( const idDict &args, idVec3 &meshOrigin, idMat3 &meshAxis, bool &poseIsSet ) = 0;
+
+
+	// Entity selection.
+	virtual void				ClearEntitySelection( void ) = 0;
+	virtual int					GetSelectedEntities( idEntity *list[], int max ) = 0;
+	virtual void				AddSelectedEntity( idEntity *ent ) = 0;
+
+	// Selection methods
+	virtual void				TriggerSelected() = 0;
+
+	// Entity defs and spawning.
+	virtual const idDict *		FindEntityDefDict( const char *name, bool makeDefault = true ) const = 0;
+	virtual void				SpawnEntityDef( const idDict &args, idEntity **ent ) = 0;
+	virtual idEntity *			FindEntity( const char *name ) const = 0;
+	virtual const char *		GetUniqueEntityName( const char *classname ) const = 0;
+
+	// Entity methods.
+	virtual void				EntityGetOrigin( idEntity *ent, idVec3 &org ) const = 0;
+	virtual void				EntityGetAxis( idEntity *ent, idMat3 &axis ) const = 0;
+	virtual void				EntitySetOrigin( idEntity *ent, const idVec3 &org ) = 0;
+	virtual void				EntitySetAxis( idEntity *ent, const idMat3 &axis ) = 0;
+	virtual void				EntityTranslate( idEntity *ent, const idVec3 &org ) = 0;
+	virtual const idDict *		EntityGetSpawnArgs( idEntity *ent ) const = 0;
+	virtual void				EntityUpdateChangeableSpawnArgs( idEntity *ent, const idDict *dict ) = 0;
+	virtual void				EntityChangeSpawnArgs( idEntity *ent, const idDict *newArgs ) = 0;
+	virtual void				EntityUpdateVisuals( idEntity *ent ) = 0;
+	virtual void				EntitySetModel( idEntity *ent, const char *val ) = 0;
+	virtual void				EntityStopSound( idEntity *ent ) = 0;
+	virtual void				EntityDelete( idEntity *ent ) = 0;
+	virtual void				EntitySetColor( idEntity *ent, const idVec3 color ) = 0;
+
+	// Player methods.
+	virtual bool				PlayerIsValid() const = 0;
+	virtual void				PlayerGetOrigin( idVec3 &org ) const = 0;
+	virtual void				PlayerGetAxis( idMat3 &axis ) const = 0;
+	virtual void				PlayerGetViewAngles( idAngles &angles ) const = 0;
+	virtual void				PlayerGetEyePosition( idVec3 &org ) const = 0;
+
+	// In game map editing support.
+	virtual const idDict *		MapGetEntityDict( const char *name ) const = 0;
+	virtual void				MapSave( const char *path = NULL ) const = 0;
```

Comment:
:ambulance: :warning: Virtual functions and default parameters should not be used together.
Virtual functions bind dynamically, but default function parameter values bind statically.
So if you implement this function in a derived class with a different default parameter, and you call it through a base pointer without parameter, it will use the default parameter specified in the base class, not the one in the derived class!